lib/fake_artist/table.ex

```diff
@@ -33,85 +53,209 @@ defmodule FakeArtist.Table do
+      }
 
-    FakeArtistWeb.Endpoint.broadcast("table:#{table_name}", "update_game", %{players: players})
+      FakeArtistWeb.Endpoint.broadcast("table:#{table_name}", "update", state)
 
-    {:reply, :ok, {id_map, table_name, user_count, state_pid}}
+      {:reply, :ok, state}
+    end
   end
 
   def handle_info(
         {:DOWN, _ref, :process, _from, _reason},
-        {id_map, table_name, user_count, _old_state_pid}
+        state = %{connected_computers: connected_computers}
       ) do
     Logger.info(fn -> "table lost connection." end)
-    Logger.info(fn -> "player count decreased from #{user_count} to #{user_count - 1}" end)
 
-    user_count = user_count - 1
+    Logger.info(fn ->
+      "player count decreased from #{connected_computers} to #{connected_computers - 1}"
+    end)
 
-    if user_count == 0 do
+    if connected_computers - 1 == 0 do
       Logger.info(fn -> "suicide" end)
-      {:stop, :shutdown, {%{}, "", 0}}
+      {:stop, :shutdown, %{}}
     else
-      {:noreply, {id_map, table_name, user_count}}
+      state = %{
+        state
+        | connected_computers: connected_computers - 1
+      }
+
+      {:noreply, state}
     end
   end
 
-  @spec assemble_players(list(), map(), number()) :: list()
-  def assemble_players(seats, id_map, active_seat) do
-    seats
-    |> Enum.with_index(0)
-    |> Enum.map(fn {id, index} ->
-      %FakeArtist.Player{
-        id: id,
-        name: Map.get(id_map, id),
-        seat: index,
-        is_active: index == active_seat
-      }
-    end)
+  @spec update_player_role(map(), number(), atom()) :: map()
+  defp update_player_role(players, id, new_role) do
+    player = Map.get(players, id)
+    player_with_new_role = %{player | role: new_role}
+    Map.put(players, id, player_with_new_role)
+  end
+
+  @spec update_player_name(map(), number(), atom()) :: map()
+  defp update_player_name(players, id, new_name) do
+    player = Map.get(players, id)
+    player_with_new_name = %{player | name: new_name}
+    Map.put(players, id, player_with_new_name)
+  end
+
+  @spec get_active_players(map(), list()) :: list()
+  defp get_active_players(players, active_players) do
+    current_active_player_id = active_players |> hd()
+    current_active_player_seat = Map.get(players, current_active_player_id).seat
+    next_seat = get_next_seat(current_active_player_seat, players |> Map.keys() |> length())
+    next_id = players |> Enum.find(fn {_, player} -> player.seat == next_seat end) |> elem(0)
+    [next_id]
+  end
+
+  @spec get_next_seat(number(), number()) :: number()
+  defp get_next_seat(current_seat, player_count) do
+    if current_seat + 1 == player_count do
+      1
+    else
+      current_seat + 1
+    end
+  end
+
+  @spec get_random_player(list()) :: tuple()
+  defp get_random_player(players) do
+    random_player = Enum.random(players)
+    without_random_player = List.delete(players, random_player)
+    {random_player, without_random_player}
```

Comment:
what about if you just do
```ex
[random_player, without_random_player ] = Enum.shuffle(players)
{random_player, without_random_player}
```
or even just return `Enum.shuffle(players)` and just change the API to return a list that you pattern match on the other side?

Actually, does the order of the `without_random_player` list matter on the other side?